### DISCUSSION TITLE
Pasting a build on a server

### DISCUSSION THREAD

**Discussion by elaineandsparky** [maruohon/litematica#334] [Q&A]
Hi. I own a server and have OP and I can paste a schematic into my server using "paste schematic in world". Possibly due to my server settings I have to change the pastecommandlimit to 5 or I get kicked for sending too many packets so it's a slower process but it works.

I just tried showing my staff how to do this and they couldn't. It kept saying in chat that they do not have permission to do this once they set the schematic off for pasting and they get kicked for spamming even when they set the pastecommandlimit to 5. They have all world edit perms and creative and i'm at a loss as to what perms they would need. They were able to paste it in when I set them as OP but this isn't a long term solution.

Does anyone know what perms someone needs to paste a schematic onto a server other than OP?

**Comment by SmylerMC** [comment]
It think Litematica uses `/setblock` on servers, so you need whatever the permission is for that.

**Comment by maruohon** [comment]
By default the mod uses the vanilla `/setblock` command, so they would need to be in creative mode and have the permission to run that command. If there is another equivalent setblock command provided by a server-side mod or plugin, that can also be used, assuming it has the same argument order. You can set the command in the Generic configs.

As for the limits.. I guess just try going even lower, to 4 or 3 commands per tick. Possibly also increasing the interval to 2 ticks, or some combination of the two. I'm still planning on adding support for also using `/fill` commands for larger areas of the same block, but sub-dividing the schematic to the "optimal"(-ish) fill and setblock commands is maybe not the simplest of things to do.

**Comment by elaineandsparky** [comment]
I've tried both these perms and it still comes up with the "you don't have perm..." error

minecraft.command.setblock
minecraft.command.minecraft:setblock
